### DISCUSSION TITLE
Running commands in sidecar containers

### DISCUSSION THREAD

**Discussion by AndreaGiardini** [actions/actions-runner-controller#499] [Questions] [answered]
Hey everyone

I've been trying to set up self hosted runners on our k8s infrastructure. The idea is to run tests for ML/AI code that is built on top of jupyter notebooks.

I managed to add a sidecar container to the runners with the image that we use to build our models. It exposes the notebook interface on localhost:8888. Now i'm trying to get the test running. The problem is that all the commands are run on the "runner" container and not on the "notebook" container.

Is there a way for the runner container to trigger commands on the sidecar container? I had a look in the README and issues but I haven't found much about it.

**Comment by mumoshu** [comment, marked as answer]
@AndreaGiardini Hey! Well sorry to say, I'm not sure how this relates to actions-runner-controller at all.

Is something running in your sidecar container exposing `localhost:8888` has some API that can be called from within your Actions workflow job that is running in the runner container?

If so, I'd suggest you to write some GitHub Actions job "step" that uses some tool to call the API against `localhost:8888`.

**Comment by mumoshu** [comment]
Why are you using sidecar containers in the first place? Can't you just `docker run` or `kubectl run` your jupyer notebook instance and feed some data from within your Actions job that is run in your runner container?

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
